### ISSUE TITLE
Inconsistent compute shader behavior between vendors with multiple samplers/SSBOs

### ISSUE THREAD

**Issue by rayanalysis** [panda3d/panda3d#1538]
## Description
I have an N-body simulation which requires both position and velocity information to be available each calculation cycle. In my first implementation, I did this by using two layout(rgba32f) uniform image3D textures set up via ComputeNode and a single dispatch_compute() call per update. This works very well on an AMD RX 6700 XT but fails on a 1050 Ti and an RTX 3060 with "Validation Error: A sampler points to a texture unit used by fixed function with an incompatible target."

I re-implemented this with a ShaderBuffer hoping to sidestep the "more than one image3D issue", however setting the velocities in an indexed SSBO like this also produces the same Validation Error.

## Steps to Reproduce
1. Access the repo here: https://github.com/rayanalysis/Ultra-Large-Scale-Structure
2. Observe program startup behavior on different vendor platforms

## Environment
Windows 10 and Ubuntu

**Comment by rdb** [comment]
Can you reproduce this with a very short sample program?

**Comment by rayanalysis** [comment]
The program is actually fairly minimal for what it does, so I don't believe I could easily reduce the complexity by much. Looking into it further, it looks like on NVIDIA the compute shader is returning NAN for every new position value -- I can "fix" it by only supplying the unmodified position data, IE not adding the velocity and acceleration component. I've experimented a bit with adjusting the float values on all the relevant variables, but haven't been able to narrow it down further than that.

**Comment by rdb** [comment]
I ran main.py with GeForce RTX 2080 Super on Windows 11 and got no error message.

**Comment by rayanalysis** [comment]
Yes, I'm not getting the error message on Windows 10 and Panda3D 1.10.13 (anymore?). However the output is still wrong on NVIDIA, just a black output after the first few frames. It should look roughly like this: https://cdn.discordapp.com/attachments/1153428034729287690/1154517106767974410/galaxy_sim_295.png

**Comment by rayanalysis** [comment]
The core issue seems to be that the compute shader is outputting NaN for the "current_position", which one can check like so:

```
        for x in range(self.size):
            for y in range(self.size):
                for z in range(self.size):
                    current_position = output_array[x, y, z, :3] # get the position from the output_array
                    print(current_position)
                    if not (np.isnan(output_array[x, y, z, :3]).any() or np.isinf(output_array[x, y, z, :3]).any()): 
                        self.grid[x][y][z] = [output_array[x, y, z, :3].tolist(), output_array[x, y, z, 3:].tolist()]
                        instance = self.instance_root.attach_new_node("Instance")
                        self.cube_model.instance_to(instance)
                        instance.set_pos(Point3(*output_array[x, y, z, :3]))
```
